File: scanner/scanner/models/season.py
from __future__ import annotations

from datetime import date
from typing import Any

from pydantic import Field

from ..utils import Language, Model
from .metadataid import SeasonId


class Season(Model):
	season_number: int
	start_air: date | None
	end_air: date | None
	external_id: dict[str, SeasonId]
	translations: dict[Language, SeasonTranslation] = {}
	extra: dict[str, Any] = Field(exclude=True)


class SeasonTranslation(Model):
	name: str | None
	description: str | None
	poster: str | None
	thumbnail: str | None
	banner: str | None

File: front/src/models/season.ts
import { z } from "zod/v4";
import { KImage } from "./utils/images";
import { zdate } from "./utils/utils";

export const Season = z.object({
	id: z.string(),
	slug: z.string(),
	seasonNumber: z.int().gte(0),
	name: z.string().nullable(),
	description: z.string().nullable(),
	entriesCount: z.int().gte(0),
	availableCount: z.int().gte(0),
	startAir: zdate().nullable(),
	endAir: zdate().nullable(),
	externalId: z.record(
		z.string(),
		z.object({
			serieId: z.string(),
			season: z.number(),
			link: z.string().nullable(),
		}),
	),

	poster: KImage.nullable(),
	thumbnail: KImage.nullable(),
	banner: KImage.nullable(),

	createdAt: zdate(),
	updatedAt: zdate(),
});

export type Season = z.infer<typeof Season>;
